map_model/src/pathfind/v2.rs

```diff
@@ -0,0 +1,146 @@
+
+use anyhow::Result;
+
+use crate::{DirectedRoadID, Map, Path, PathRequest, PathStep, TurnID};
+
+/// Transform a sequence of roads representing a path into the current lane-based path, by picking
+/// particular lanes and turns to use.
+pub fn path_v2_to_v1(req: PathRequest, road_steps: Vec<DirectedRoadID>, map: &Map) -> Result<Path> {
+    // This is a somewhat brute-force method: run Dijkstra's algorithm on a graph of lanes and
+    // turns, but only build the graph along the path of roads we've already found. This handles
+    // arbitrary lookahead needed, and forces use of the original start/end lanes requested.
+    //
+    // Eventually we'll directly return road-based paths. Most callers will actually just use that
+    // directly, and mainly the simulation will need to expand to specific lanes, but it'll do so
+    // dynamically/lazily to account for current traffic conditions.
+    let mut graph = petgraph::graphmap::DiGraphMap::new();
+    for pair in road_steps.windows(2) {
+        for src in pair[0].lanes(req.constraints, map) {
+            for dst in pair[1].lanes(req.constraints, map) {
+                let turn = TurnID {
+                    parent: map.get_l(src).dst_i,
+                    src,
+                    dst,
+                };
+                if map.maybe_get_t(turn).is_some() {
+                    graph.add_edge(src, dst, turn);
+                }
+            }
+        }
+    }
+
+    match petgraph::algo::astar(
+        &graph,
+        req.start.lane(),
+        |l| l == req.end.lane(),
+        // TODO We could include the old lane-changing penalties here, but I'm not sure it's worth
+        // the complication. The simulation layer will end up tuning those anyway.
+        |_| 1,
+        |_| 0,
+    ) {
+        Some((_, path)) => {
+            let mut steps = Vec::new();
+            for pair in path.windows(2) {
+                steps.push(PathStep::Lane(pair[0]));
+                // We don't need to look for this turn in the map; we know it exists.
+                steps.push(PathStep::Turn(TurnID {
+                    parent: map.get_l(pair[0]).dst_i,
+                    src: pair[0],
+                    dst: pair[1],
+                }));
+            }
+            steps.push(PathStep::Lane(req.end.lane()));
+            assert_eq!(steps[0], PathStep::Lane(req.start.lane()));
+            // TODO No uber-turns yet!
+            Ok(Path::new(map, steps, req, Vec::new()))
+        }
+        None => bail!(
+            "path_v2_to_v1 found road-based path, but not a lane-based path matching it for {}",
+            req
+        ),
+    }
+}
+
+// TODO This is an attempt that looks at windows of 2 roads at a time to pick particular lanes and
+// turns. It doesn't work in most cases with multiple lane choices -- I think we need at least a
+// window of 3 roads. I'll write that function in the future, for the simulation layer to use
+// "lazily".
+fn _broken_path_v2_to_v1(
```

Comment:
FYI - I didn't really review this function, since it seems like you're just stashing some draft code here.

Reply:
Sounds good -- yeah, I just want it under version control for the record.